include/multipass/settings/dynamic_setting_spec.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2021-2022 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef MULTIPASS_DYNAMIC_SETTING_SPEC_H
+#define MULTIPASS_DYNAMIC_SETTING_SPEC_H
+
+#include "basic_setting_spec.h"
+
+#include <functional>
+
+namespace multipass
+{
+class DynamicSettingSpec : public multipass::BasicSettingSpec
```

Comment:
Maybe `CustomSettingSpec`? Otherwise it suggests it could change over a lifetime of the spec?

Reply:
"Dynamic" here refers to something that varies at runtime, as opposed to that which can be established at compile time (often "static"). It is a usual meaning for the word in programming contexts, e.g. `dynamic_cast`, dynamic memory allocation, dynamic binding, a container that grows dynamically, etc. In this case, the `SettingSpec`'s behavior is determined dynamically, according to a callable provided at runtime. 

That said, I am fine with `CustomSettingSpec` too, so will change to that since you prefer it.